app/model/repository/Groups.php

```diff
@@ -144,6 +150,54 @@ private function fetchGroupsByMembership(User $user, bool $admins): array
         return $res;
     }
 
+    /**
+     * Find immediate children of all parent groups on a given list.
+     * This can be used to iteratively get whole subtree of groups quite efficiently.
+     * @param string[] $parentsIds list of IDs of all parent groups
+     * @return Group[]
+     */
+    private function findAllChildrenOf(array $parentsIds): array
+    {
+        if (!$parentsIds) {
+            return [];
+        }
+        $qb = $this->createQueryBuilder('g'); // takes care of softdelete cases
+        $qb->andWhere($qb->expr()->in("g.parentGroup", $parentsIds));
+        return $qb->getQuery()->getResult();
+    }
+
+    /**
+     * Find all groups in are supervised by a user (as supervisor, admin, or admin inherently).
```

Comment:
A bit weird formulation, `in` does not seem to fit there...

Reply:
Fixed.